internal/tasks/account_management.go

```diff
@@ -0,0 +1,166 @@
+	}
+
+	for _, accountName := range managedAccountNames {
+		if !slices.Contains(existingAccountNames, accountName) {
+			accountNames = append(accountNames, accountName)
+		}
+	}
+
+	var toEnforceAccountNames []string
+	_, err = j.db.Select(&toEnforceAccountNames, "SELECT name FROM accounts WHERE is_managed = true and next_account_enforcement_at < $1", j.timeNow())
+	if err != nil {
+		return nil, err
+	}
+	accountNames = append(accountNames, toEnforceAccountNames...)
+
+	if len(accountNames) == 0 {
+		return nil, sql.ErrNoRows
+	}
+
+	return accountNames, nil
+}
+
+func (j *Janitor) enforceManagedAccounts(ctx context.Context, accountNames []string, labels prometheus.Labels) error {
+	for _, accountName := range accountNames {
+		account, securityScanPolicies, err := j.amd.ConfigureAccount(models.Account{Name: accountName})
+		if err != nil {
+			return err
+		}
+
+		if account == nil {
+			accountModel, err := keppel.FindAccount(j.db, accountName)
+			if err != nil {
+				return err
+			}
+			// assume the account got already deleted
+			if errors.Is(err, sql.ErrNoRows) {
+				return nil
+			}
+
+			_, err = j.db.Exec("UPDATE accounts SET in_maintenance = TRUE WHERE name = $1", accountName)
+			if err != nil {
+				return err
+			}
+			// avoid quering the account twice and we set this field just above via SQL
+			accountModel.InMaintenance = true
+
+			resp, err := j.processor().DeleteAccount(ctx, *accountModel)
+			if err != nil {
+				return err
+			}
+			if resp != nil {
+				logg.Error("Deleting account %s failed: %s", accountName, resp.Error)
+				return errors.New(resp.Error)
+			}
+		} else {
+			userIdentity := janitorUserIdentity{TaskName: "account-management"}
+
+			getSubleaseToken := func(peer models.Peer) (string, *keppel.RegistryV2Error) {
+				viewScope := auth.Scope{
+					ResourceType: "keppel_account",
+					ResourceName: account.Name,
+					Actions:      []string{"view"},
+				}
+
+				client, err := peerclient.New(ctx, j.cfg, peer, viewScope)
+				if err != nil {
+					return "", keppel.AsRegistryV2Error(err)
+				}
+
+				subleaseToken, err := client.GetSubleaseToken(ctx, account.Name)
+				if err != nil {
+					return "", keppel.AsRegistryV2Error(err)
+				}
+				return subleaseToken, nil
+			}
+
+			jsonBytes, err := json.Marshal(securityScanPolicies)
+			if err != nil {
+				return err
+			}
+
+			setCustomFields := func(account *models.Account) error {
+				account.IsManaged = true
+				account.SecurityScanPoliciesJSON = string(jsonBytes)
+				nextAt := j.timeNow().Add(j.addJitter(1 * time.Hour))
+				account.NextAccountEnforcementAt = &nextAt
+				return nil
+			}
+
+			_, rerr := j.processor().CreateOrUpdateAccount(ctx, *account, userIdentity.UserInfo(), janitorDummyRequest, getSubleaseToken, setCustomFields)
+			if rerr != nil {
+				return rerr
+			}
+		}
+
+		_, err = j.db.Exec(accountAnnouncementDoneQuery, accountName, j.timeNow().Add(j.addJitter(1*time.Hour)))
+		if err != nil {
+			return err
+		}
```

Comment:
This is the wrong query. You're updating `NextFederationAnnouncementAt` instead of `NextAccountEnforcementAt` here. I assume this is a copy-paste error. If so, please do a sweep if there are other copy-paste errors like this that I did not catch.